### DISCUSSION TITLE
Query to pull timezone for a GeoPoint on a given day

### DISCUSSION THREAD

**Discussion by luckynipun** [RomanIakovlev/timeshape#91] [Q&A] [answered]
Hi Team,

Thank you for building such a cool library.

My requirement is to get the timezone for a given GeoPoint. This query returns the timezone information for the GeoPoint considering current day by default.
But I need accurate timezone information when I query the it for older dates as well.

Given that timezone offset change based on dates throughout the year, is it possible to get the timezone for a given date?

I apologise, I didn't look deep into code, hence the question.
Is this an easy addition with the existing code? or a big feature to develop?

**Comment by RomanIakovlev** [comment, marked as answer]
Hi,

It’s great you’re enjoying using Timeshape so far. As for the time zones, they don’t change that often. You probably mix them up with the daylight saving time? 

The information about which physical time instant corresponds to local time at a given location is encoded in database, which is distributed together with JRE (it’s called ‘tzdb’). If you use ‘ZoneId’ you get from Timeshape to construct, say, ’ZonedDateTime’, you can find which time at the given instant is observed in the given location, using the ‘tzdb’ information. It’s all done by Java, you just have to make sure to use the classes from ‘java.time.*’ that match your use case (‘LocalDateTime’ or ‘ZonedDateTime’), and it just works, giving you the correct time year round, future and past.

Does this answer your question?

**Reply by luckynipun** [reply]
Hi Roman,

Thank you.
You are right, I mixed up timezone with daylight saving time.

After looking into ZonedDateTime in detail, I understand what you have explained.

Thanks again for spending time and answering my ignorant query.
